tests/models/opt/test_modeling_opt.py

```diff
@@ -0,0 +1,494 @@
+        return GPT2Tokenizer.from_pretrained("patrickvonplaten/opt_gpt2_tokenizer")
+
+    @slow
+    def test_inference_no_head(self):
+        model = OPTModel.from_pretrained("ArthurZ/350-m").to(torch_device)
+        input_ids = _long_tensor([[0, 31414, 232, 328, 740, 1140, 12695, 69, 46078, 1588, 2]])
+        attention_mask = input_ids.ne(model.config.pad_token_id)
+        with torch.no_grad():
+            output = model(input_ids=input_ids, attention_mask=attention_mask).last_hidden_state
+        expected_shape = torch.Size((1, 11, 1024))
+        self.assertEqual(output.shape, expected_shape)
+        expected_slice = torch.tensor(
+            [[0.7144, 0.8143, -1.2813], [0.7144, 0.8143, -1.2813], [-0.0467, 2.5911, -2.1845]], device=torch_device
+        )
+        self.assertTrue(torch.allclose(output[:, :3, :3], expected_slice, atol=1e-3))
+
+
+@require_tokenizers
+@require_torch
+@slow
+class OPTEmbeddingsTest(unittest.TestCase):
+    def setUp(self):
+        super().setUp()
+        self.path_model = "ArthurZ/opt-350m"
+
+    def test_load_model(self):
+        try:
+            _ = OPTForCausalLM.from_pretrained(self.path_model)
+        except BaseException:
+            self.fail("Failed loading model")
+
+    @require_torch
+    @torch.no_grad()
+    def test_logits(self):
+        model = OPTForCausalLM.from_pretrained(self.path_model)
+        model = model.eval()
+        # tokenizer = GPT2Tokenizer.from_pretrained("patrickvonplaten/opt_gpt2_tokenizer")
+        tokenizer = GPT2Tokenizer.from_pretrained("patrickvonplaten/opt_gpt2_tokenizer")
+        tokenizer.add_special_tokens({"pad_token": "<pad>"})
+
+        prompts = [
+            "Today is a beautiful day and I want to",
+            "In the city of",
+            "Paris is the capital of France and",
+            "Computers and mobile phones have taken",
+        ]
+        input_ids = tokenizer(prompts, return_tensors="pt", padding=True).input_ids
+        logits = model(input_ids)[0].mean(dim=-1)
+        # logits_meta = torch.load(self.path_logits_meta)
+        logits_meta = torch.Tensor(
+            [
+                [1.3851, -13.8923, -10.5229, -10.7533, -0.2309, -10.2384, -0.5365, -9.0947, -5.1670],
+                [-4.7073, -10.6276, -3.9415, -21.5242, -0.2822, -0.2822, -0.2822, -0.2822, -0.2822],
+                [0.6247, -3.4229, -8.9179, -1.4297, -14.1650, 1.4146, -9.0218, -0.2703, -0.2703],
+                [6.4783, -1.9913, -10.7926, -2.3336, 1.5092, -0.9974, -6.8213, 1.3477, 1.3477],
+            ]
+        )
+
+        assert torch.allclose(logits, logits_meta, atol=1e-4)
+
+
+@require_tokenizers
+@slow
+class OPTGenerationTest(unittest.TestCase):
+    def setUp(self):
+        super().setUp()
+        self.path_model = "ArthurZ/opt-350m"
+        # self.path_logits_meta = "/home/younes/Desktop/Work/metaseq-conversion/logits_metaseq_gpt2_tokenizer.p"  # TODO add the logits somewhere?
+
+    def test_generation(self):
```

Comment:
Let's add 3 more generation tests here with the 125m, 350m and 1300m checkpoint that all show sensible outputs for generation

Reply:
Good idea, working on it